### PR TITLE
feat: add ValidateInput override to JoinNode to validate individual predecessor outputs

### PR DESCRIPTION
**Pull request by hanorik**
This PR adds a `ValidateInput` override to `JoinNode` to validate the outputs of individual predecessor nodes against the node's input schema, rather than validating the aggregated map as a single value.

### Key Changes
- **`JoinNode.ValidateInput` implementation**:
  - Overrides the default base node input validation.
  - If the input is a `map[string]any` containing outputs from multiple predecessor nodes, it validates each predecessor's output value individually against `InputSchema()`.
  - Returns a map of validated values or returns a wrapped validation error including the predecessor's name (formatted as `predecessor "X": <inner error>`).
  - If the input is not a `map[string]any` (e.g., a raw single predecessor value), it falls back to `BaseNode.ValidateInput`.
  - If the node's `InputSchema()` is `nil`, the input is returned as-is.

### Automated Tests
Added unit tests in [workflow/join_node_test.go](file:///usr/local/google/home/hanorik/adk-go/workflow/join_node_test.go) to verify:
- **All match schema**: Validation succeeds when three predecessor values all match the schema, returning the map of validated values.
- **One fails validation**: Returns a detailed error matching the format `predecessor "X": <inner error>` when one of three predecessor values fails validation.
- **Not a map falls back**: Correctly falls back to `BaseNode.ValidateInput` when the input is a raw value rather than a `map[string]any`.
- **Nil schema returns as-is**: Simply returns the input as-is when `InputSchema()` is `nil`.

### CODE REVIEW

File: workflow/join_node.go [hunk 1]
```diff
@@ -49,3 +50,26 @@ func (n *JoinNode) Run(ctx agent.InvocationContext, input any) iter.Seq2[*sessio
 		yield(event, nil)
 	}
 }
+
+// ValidateInput overrides BaseNode.ValidateInput. Instead of validating the
+// aggregated map as a single value, it validates each predecessor's output
+// individually against the node's InputSchema.
+func (n *JoinNode) ValidateInput(input any) (any, error) {
+	schema := n.InputSchema()
```
Comment (wolo-lab) [inline workflow/join_node.go:72]: We could add a new constructor so creating a new JoinNode with schemas will be simpler than:

```
	joinWithSchema := &JoinNode{
		BaseNode: NewBaseNodeWithSchemas("join", "", NodeConfig{}, stringSchema, nil),
	}
```

in a new constructor please add a doc explaining how schema is applied (there are two possible interpretations: schema for each value in the map (recommended) or schema for a map. Add also a note how nil in the map is handled.

File: workflow/join_node_test.go [hunk 2]
```diff
@@ -229,3 +232,86 @@ func inputRecorder(name string, seen *any) *FunctionNode {
 			return "done", nil
 		}, defaultNodeConfig)
 }
+
+func TestJoinNode_ValidateInput(t *testing.T) {
```
Comment (wolo-lab) [inline workflow/join_node_test.go:236]: please add a test for nil in the map and empty map.

**Review by wolo-lab — APPROVED** [review]

File: workflow/join_node.go (hunk 1 above)
Comment (hanorik) [reply workflow/join_node.go:72]: Done.

File: workflow/join_node_test.go (hunk 2 above)
Comment (hanorik) [reply workflow/join_node_test.go:236]: Done.
